input/icinga2.go

```diff
@@ -0,0 +1,358 @@
+	StateType              string      `json:"state_type"`
+	LastState              string      `json:"last_state"`
+	LastStateType          string      `json:"last_state_type"`
+	LastHardState          json.Number `json:"last_hard_state"`
+	Output                 string      `json:"output"`
+	WebURL                 string      `json:"url"`
+	LastStateChangeStr     json.Number `json:"last_state_change"`
+	LastHardStateChangeStr json.Number `json:"last_hard_state_change"`
+}
+
+func (h Host) ColoredState() string {
+	return ColorHostState(h.State);
+}
+
+func (h Host) ColoredLastState() string {
+	return ColorServiceState(h.LastState);
+}
+
+func (h Host) LastStateChange() time.Time {
+	return JsonToTime(h.LastStateChangeStr);
+}
+
+func (h Host) LastHardStateChange() time.Time {
+	return JsonToTime(h.LastHardStateChangeStr);
+}
+
+func (h Host) AgoString() string {
+	return AgoString(h.LastStateChange());
+}
+
+
+type Service struct {
+	CheckAttempt           float32     `json:"check_attempt"`
+	Name                   string      `json:"name"`
+	DisplayName            string      `json:"display_name"`
+	State                  string      `json:"state"`
+	StateType              string      `json:"state_type"`
+	LastState              string      `json:"last_state"`
+	LastStateType          string      `json:"last_state_type"`
+	LastHardState          json.Number `json:"last_hard_state"`
+	Output                 string      `json:"output"`
+	WebURL                 string      `json:"url"`
+	LastStateChangeStr     json.Number `json:"last_state_change"`
+	LastHardStateChangeStr json.Number `json:"last_hard_state_change"`
+}
+
+func (s Service) ColoredState() string {
+	return ColorServiceState(s.State);
+}
+
+func (s Service) ColoredLastState() string {
+	return ColorServiceState(s.LastState);
+}
+
+func (s Service) LastStateChange() time.Time {
+	return JsonToTime(s.LastStateChangeStr);
+}
+
+func (s Service) LastHardStateChange() time.Time {
+	return JsonToTime(s.LastHardStateChangeStr);
+}
+
+func (s Service) AgoString() string {
+	return AgoString(s.LastStateChange());
+}
+
+
+type Notification struct {
+	Author    string      `json:"author"`
+	Comment   string      `json:"comment"`
+	Target    string      `json:"target"`
+	Type      string      `json:"type"`
+	Timestamp json.Number `json:"timet"`
+	DateTime  string      `json:"long_date_time"`
+	Host      Host        `json:"host"`
+	Service   Service     `json:"service"`
+	Channels  []string    `json:"channels"`
+}
+
+
+type Icinga2Module struct {
+	channelMapping hgmapping
+	channel        chan IRCMessage
+}
+
+type hgmapping struct {
+	DefaultChannel    string              `mapstructure:"default"`
+	HostGroupMappings map[string][]string `mapstructure:"hostgroups"`
+	ExplicitMappings  map[string][]string `mapstructure:"explicit"`
+}
+
+func (m *Icinga2Module) Init(c *viper.Viper, channel *chan IRCMessage) {
+	err := c.UnmarshalKey("default", &m.channelMapping.DefaultChannel)
```

Comment:
You don't need three unmarshalKey() calls and can use just one like for the GitLab module https://github.com/fleaz/CptHook/blob/master/input/gitlab.go#L37

Reply:
The problem with unmarshal() is that keys with dots aren't working but we need them for hostnames. 
https://github.com/fleaz/CptHook/pull/31#issuecomment-436089074

Reply:
Ah yes, there was an issue. I remember that. Still unfixed upstream :(